avm/src/vm/opcode.rs

```diff
@@ -1717,14 +1717,16 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
   io!("findl", |args, mem| {
     Box::pin(async move {
       let mut hand_mem = mem.write().await;
-      let arr = hand_mem.read_fractal(args[0]);
+      let arr = hand_mem.read_fractal(args[0]).to_vec();
       let len = arr.len();
       let subhandler = HandlerFragment::new(args[1], 0);
       for i in 0..len {
-        let mut hm = hand_mem.fork();
+        let mut hm = hand_mem.clone();
         hm.set_addr(CLOSURE_ARG_MEM_START + 1, arr[i].0, arr[i].1 as usize);
         hm = subhandler.clone().run(hm).await;
-        let val = hm.read_fixed(CLOSURE_ARG_MEM_START);
+        // The sequential version of `each` is allowed to have side-effects
```

Comment:
`find`?

this also opens the non-blocking question of whether `map` should be allowed to have side-effects?

Reply:
Oh, yes, let me fix that.

And `find` and `map` can't do that because they run in parallel and mutating outer scope would be nondeterministic.

That's what the `pure` flag on functions is about -- to try to track that and auto-downgrade to these linear opcodes when we can't safely run code in parallel -- but that's also not working yet, hence the `eachLin`, `findLin`, etc being directly exposed (though I'm also not opposed to keeping those exposed long-term for other reasons (like predictable execution ordering when communicating to some other server, for instance).